genai_stack/retriever/langchain.py

```diff
@@ -1,20 +1,69 @@
-from typing import List
-from langchain.docstore.document import Document
+from .base import BaseRetrieverConfigModel, BaseRetrieverConfig, BaseRetriever
+from genai_stack.retriever.utils import parse_search_results
 
-from .base import BaseRetriever
+
+class LangChainRetrieverConfigModel(BaseRetrieverConfigModel):
+    """
+    Data Model for the configs
+    """
+    pass
+
+
+class LangChainRetrieverConfig(BaseRetrieverConfig):
+    data_model = LangChainRetrieverConfigModel
 
 
 class LangChainRetriever(BaseRetriever):
-    required_fields = []
+    config_class = LangChainRetrieverConfig
+
+    def retrieve(self, query:str): 
+
+        prompt_template = self.get_prompt(query=query)
+
+        prompt_dict = {
+            "query": query
+        }
+
+        if "context" in prompt_template.input_variables:
+            prompt_dict['context'] = self.get_context(query=query)
+        
+        if "history" in prompt_template.input_variables:
+            prompt_dict['history'] = self.get_chat_history()
+
+        final_prompt_template =  prompt_template.template.format(
+            **{k:v for k,v in prompt_dict.items()}
+        )
+
+        response = self.mediator.get_model_response(prompt=final_prompt_template)
+
+        if "history" in prompt_template.input_variables:
```

Comment:
Move them to the mediator like self.mediator.add_to_memory and do the check there like

if self._is_component_available("memory"):
   self._stack.memory.add_text(...)